pkg/client/sc/watch.go

```diff
@@ -0,0 +1,47 @@
+package sc
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/apache/servicecomb-service-center/server/core"
+	pb "github.com/apache/servicecomb-service-center/server/core/proto"
+	scerr "github.com/apache/servicecomb-service-center/server/error"
+	"github.com/gorilla/websocket"
+)
+
+const (
+	apiWatcherURL        = "/v4/%s/registry/microservices/%s/watcher"
+	apiListAndWatcherURL = "/v4/%s/registry/microservices/%s/listwatcher"
+)
+
+func (c *SCClient) Watch(ctx context.Context, domainProject, serviceId string, callback func(*pb.WatchInstanceResponse)) *scerr.Error {
```

Comment:
please rename the serviceId to selfServiceId, because it is not the target serviceId here

Reply:
Modified, please review